extra/classes/prettyprinter/prettyprinter.factor

```diff
@@ -0,0 +1,91 @@
+! Copyright (C) 2023 Jean-Marc Lugrin.
+! See https://factorcode.org/license.txt for BSD license.
+
+! Print a class heierarchy in the listener
+! or to a file: "TESTDUMP.TXT" utf8 [ gadget hierarchy. ] with-file-writer
+
+
+USING:  classes hashtables ui.gadgets assocs kernel sequences prettyprint  vectors 
+math io formatting strings sorting accessors io.styles vocabs ;
+
+IN: classes.prettyprinter
+
+ERROR: not-a-class-error ;
+
+<PRIVATE
+
+CONSTANT: in-col 40
+
+: add-child  ( c h  -- )
+    over ! ( c h -- c h c )
```

Comment:
it is often useful to have stack effect comments like this as you learn factor and/or build something complex, but then we often remove them before pushing unless the stack effect comment is particularly useful for some internal non-obvious quotation